plugins/flatpak/gs-flatpak.c

```diff
@@ -1388,10 +1454,22 @@ gs_flatpak_add_updates_pending (GsFlatpak *self, GsAppList *list,
 			g_warning ("failed to add flatpak: %s", error_local->message);
 			continue;
 		}
-		gs_app_set_state (app, AS_APP_STATE_UPDATABLE_LIVE);
+
+		if (gs_app_get_kind (app) == AS_APP_KIND_RUNTIME)
+			main_app = get_main_app_of_related (self, app, cancellable,
+							    &error_local);
+
+		if (main_app == NULL)
+			main_app = app;
```

Comment:
Maybe print `error_local->message` with a `g_warning()` here would be good as well

Reply:
I will do it but only when the error is not `...NO_DATA` because not all runtimes are extensions.

Reply:
Ok, but then a comment in the code explaining why that exception would be useful